identity/handler.go

```diff
@@ -916,6 +917,19 @@ func (h *Handler) patch(w http.ResponseWriter, r *http.Request, ps httprouter.Pa
 
 	patchedIdentity := WithAdminMetadataInJSON(*identity)
 
+	// We need to check for all sub-paths of /credentials, since jsonx.ApplyJSONPatch
+	// only checks full paths.
+	for _, path := range gjson.GetBytes(requestBody, "#.path").Array() {
+		if strings.HasPrefix(path.String(), "/credentials/") {
+			h.r.Writer().WriteError(w, r, errors.WithStack(
+				herodot.
+					ErrBadRequest.
+					WithReasonf("An error occured when applying the JSON patch").
+					WithErrorf("patch includes denied sub-path of /credentials: %s", path.String())))
+			return
+		}
+	}
+
 	if err := jsonx.ApplyJSONPatch(requestBody, &patchedIdentity, "/id", "/stateChangedAt", "/credentials"); err != nil {
```

Comment:
Does this only do full string match? E.g `/credentials/abc` works?

Reply:
Yeah, we could also fix it in `jsonx`, or extend `jsonx` to take path globs, probably, which makes the behaviour more explicit.

Then at the call site we whould have

```go
jsonx.ApplyJSONPatch(requestBody, &patchedIdentity, "/id", "/stateChangedAt", "/credentials", "/credentials/**);
```